content/en/blog/2023/ambient-ebpf-redirection/index.md

```diff
@@ -0,0 +1,72 @@
+
+Furthermore, eBPF enables deeper visibility and additional context for packets in the kernel, allowing for more efficient and flexible management of data flow compared with iptables.
+
+## How it works
+
+An eBPF program, attached to the [traffic control](https://man7.org/linux/man-pages/man8/tc-bpf.8.html) ingress and egress hook, has been compiled into the Istio CNI component. `istio-cni` will watch pod events and attach/detach the eBPF program to other related network interfaces when the pod is moved into or out of ambient mode.
+
+Using an eBPF program (instead of iptables) eliminates the need to encapsulate tasks (for Geneve), allowing the routing tasks to be customized in kernel space instead. This yields both gains in performance, and additional flexibility, in routing.
+
+{{< image width="55%"
+    link="ambient-ebpf.png"
+    caption="ambient eBPF architecture"
+    >}}
+
+All traffic to/from the application pod will be intercepted by eBPF and redirected to the corresponding ztunnel pod. On the ztunnel side, proper redirection will be performed based on connection lookup results within the eBPF program. This provides more efficient control of the network traffic between the application and ztunnel.
+
+## How to enable eBPF redirection in Istio ambient mode
+
+Follow the [Getting Started with Ambient Mesh](/blog/2022/get-started-ambient/) instructions to set up your cluster but, when you install Istio, set the `values.cni.ambient.redirectMode` configuration parameter to `ebpf`:
+
+{{< text bash >}}
+$ istioctl install --set profile=ambient --set values.cni.ambient.redirectMode="ebpf"
+{{< /text >}}
+
+Check the `istio-cni` logs to confirm eBPF redirection is on:
+
+{{< text plain >}}
+ambient Writing ambient config: {"ztunnelReady":true,"redirectMode":"eBPF"}
+{{< /text >}}
+
+## Performance gains
+
+The latency and throughput (QPS) for eBPF mode is somewhat better than IPtables mode. The following tests were run in a kind cluster with
+a Fortio client sending requests to a Fortio server, both running in ambient mode (with debug log turned off in eBPF) and on the same k8s node.
+
+{{< image width="90%" link="./MaxQPS.png" alt="Max QPS with varying number of connections" title="Max QPS with varying number of connections" caption="Max QPS with varying number of connections" >}}
+
+The above metrics were produced with the following command:
+
+{{< text bash >}}
+$ fortio load -t 60s -qps 0 -c <num_connections> http://<fortio-svc-name>:8080
+{{< /text >}}
+
+{{< image width="90%" link="./Latency-with-8000-qps.png" alt="Latency (ms) for QPS 8000 with varying number of connections" title="Latency(ms) for QPS 8000 with varying number of connections" caption="Latency (ms) for QPS 8000 with varying number of connections" >}}

```

Comment:
Latency metrics should have at what percentile the data is at

Reply:
OK. Current latency metrics is based on average latency. Do you think we should more precise like P50/P75/P90/P99 metrics for latency dimension?

Reply:
yes, "latency" is not a useful metric without the associated percentile. We can just put in the doc which one we used, we don't need to report multiple